File: SRC/BE/food-delivery-service/src/main/java/com/trung/fooddeliveryservice/dto/response/RestaurantResponse.java
package com.trung.fooddeliveryservice.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.trung.fooddeliveryservice.util.enums.RestaurantStatus;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RestaurantResponse implements Serializable {
    private Long id;
    private Long ownerId;
    private String name;
    private String phone;
    private String address;
    private Double latitude;
    private Double longitude;
    private String imageUrl;
    private RestaurantStatus status;
    private String openTime;
    private String closeTime;
    private Double rating;
    private Integer reviewCount;

    @JsonProperty("totalReviews")
    public Integer getTotalReviews() {
        return reviewCount != null ? reviewCount : 0;
    }
    private Boolean isLocked;
    private String lockedReason;
    private LocalDateTime lockedAt;
    private List<MenuItemResponse> menuItems;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: SRC/APP/app/src/main/java/com/example/omnigo/features/customer/food/data/remote/dto/RestaurantResponse.kt
package com.example.omnigo.features.customer.food.data.remote.dto

data class RestaurantResponse(
    val id: Long,
    val ownerId: Long?,
    val name: String,
    val phone: String?,
    val address: String?,
    val latitude: Double?,
    val longitude: Double?,
    val imageUrl: String?,
    val status: String?,
    val openTime: String?,
    val closeTime: String?,
    val rating: Double?,
    val reviewCount: Int?,
    val totalReviews: Int?,
    val isLocked: Boolean?,
    val lockedReason: String?,
    val lockedAt: String?,
    val menuItems: List<MenuItemResponse>?,
    val createdAt: String?,
    val updatedAt: String?
)
